Marginalia fails for certain cascalog op definitions

I've finally pinned this down! I'm using marginalia on a [cascalog](https://github.com/nathanmarz/cascalog) project, which requires that some custom operations be defined like this, with a vector instead of a symbol in the normal symbol position. (This is required for cascalog to allow static arguments as well as dynamic arguments.)

```
(defbufferop [sparse-vec [length missing-val]]
    [tuples]
    [[(sparse-vector length missing-val tuples)]])
```

If I run marginalia on any namespace containing some definition like this, I get this error:

```
Caused by: java.lang.ClassCastException: clojure.lang.PersistentVector cannot be cast to clojure.lang.Symbol
```

Would be it be possible to catch that exception for cases like this?

Thanks a lot, guys!
Sam

--- a/src/marginalia/parser.clj
+++ b/src/marginalia/parser.clj
@@ -153,11 +153,25 @@
   [form raw nspace-sym]
   [nil raw nspace-sym])
 
+(defn dispatch-inner-form
+  [form raw nspace-sym]
+  (conj
+   (reduce (fn [[adoc araw] inner-form]
+             (if (seq? inner-form)
+               (let [[d r] (dispatch-form inner-form
+                                          araw
+                                          nspace-sym)]
+                 [(str adoc d) r])
+               [adoc araw]))
+           [nil raw]
+           form)
+   nspace-sym))
+
 (defmethod dispatch-form :default
   [form raw nspace-sym]
   (if (re-find #"^def" (-> form first name))
     (extract-common-docstring form raw nspace-sym)
-    [nil raw nspace-sym]))
+    (dispatch-inner-form form raw nspace-sym)))
 
 (defn extract-docstring [m raw nspace-sym]
   (let [raw (join "\n" (subvec raw (-> m :start dec) (:end m)))

